### PR TITLE
psf: fix getting a relative directory error on windows

### PR DESCRIPTION
**Pull request by Greedysky**
Fix getting a relative directory error on windows, which causes the file(such as .minidsf .minipsf .miniqsf .minissf) not to be imported

### CODE REVIEW

**Comment by kuba160** [comment]
DeaDBeeF does converting to forward slash internally, so I looked what causes the problems.

In `eng_psf.c`:
```c
void
ao_getlibpath (const char *path, const char *libname, char *libpath, int size) {
    const char *e = strrchr (path, ':');
    if (!e) {
        e = strrchr (path, '/');
    }
    if (e) {
    ...
```
It checks first if path includes `:` symbol and uses it to separate between file and filepath. If it doesn't it just uses forward slash. I have no idea why it checks for `:`, but it clearly breaks on windows paths. Having it to always use forward slash (`if (1)`) fixes it for local files, but it didn't work for samba paths (starting with `//`).

**Comment by kuba160** [comment]
my mistake, I was missing `.psflib` file on samba disk, it works just fine on them. Never used psf before 🙂.

**Review by Alexey-Yakovenko — CHANGES_REQUESTED** [review]

File: plugins/psf/psfmain.c [hunk 1]
```diff
@@ -260,3 +260,20 @@ int
 ao_command (uint32 type, void *handle, int32 command, int32 param) {
 	return (*types[type].command)(handle, command, param);
 }
+
+void
+ao_getlibpath (const char *path, const char *libname, char *libpath) {
+    const char *e = strrchr (path, 0x5C); // '\\'
```
[the remark this answers is not in the document]
Comment (Greedysky) [reply plugins/psf/psfmain.c:266]: I started with '\\'，but there seemed to be a problem on ubuntu 16.04 with gcc 5.4, so I used the ASCII code directly.

File: plugins/psf/psfmain.c [hunk 2]
```diff
@@ -260,3 +260,20 @@ int
 ao_command (uint32 type, void *handle, int32 command, int32 param) {
 	return (*types[type].command)(handle, command, param);
 }
+
+void
+ao_getlibpath (const char *path, const char *libname, char *libpath) {
```
[the remark this answers is not in the document]
Comment (Greedysky) [reply plugins/psf/psfmain.c:265]: Yes, I agree with you.

**Comment by Alexey-Yakovenko** [comment]
@Greedysky are you intending to fix the requested changes in this PR?

**Comment by Greedysky** [comment]
> 
> 
> @Greedysky are you intending to fix the requested changes in this PR?

Done.

**Comment by Alexey-Yakovenko** [comment]
Thanks!

**Pull request merged**